UnityWeld/Binding/Internal/TypeResolver.cs

```diff
@@ -9,11 +9,43 @@
 
 namespace UnityWeld.Binding.Internal
 {
+    /// <summary>
+    /// ViewModel provider delegate
+    /// </summary>
+    /// <param name="component"></param>
+    /// <returns>Tuple with ViewModel name and ViewModel itself</returns>
+    public delegate ViewModelProviderData ViewModelProviderDelegate(Component component);
+
+    /// <summary>
+    /// Contains ViewModel data
+    /// </summary>
+    public class ViewModelProviderData 
+    {
+        public readonly object Model;
+        public readonly string TypeName;
+
+        public ViewModelProviderData(object model, string typeName)
+        {
+            Model = model;
+            TypeName = typeName;
+        }
+    }
+
     /// <summary>
     /// Helper class for setting up the factory for use in the editor.
     /// </summary>
     public static class TypeResolver
     {
+        private static readonly IList<Type> BindingAttributeTypes = new List<Type>(2)

```

Comment:
Why does this and ViewModelProviders both have a size of 2 but only one element? Since most users aren't going to register a custom binding attribute I'd rather just have this default to one, and afaik adding an element to a .NET list is very cheap.

Reply:
The thought process is to avoid additional allocations after start. And since in most cases we need list with size of 2 (max, default and custom) I've decided to use this number as initial capacity.
But I missed the fact that it happens only during initialization phase.
However I wouldn't use default capacity, it will be 3 useless refs in memory during whole app life time, unacceptable! :)
What do you think?